examples/simple_mera/simple_mera.py

```diff
@@ -0,0 +1,318 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Very simple scale-invariant MERA.
+
+Uses automatic differentiation to construct ascending and descending
+superoperators, as well as environment tensors so that only a single tensor
+network needs to be defined: The network for computing the energy using a
+single layer of MERA together with a reduced state and a hamiltonian.
+
+Run as the main module, this module executes a script that optimizes a MERA
+for the critical Ising model.
+
+For the scale-invairant MERA, see arXiv:1109.5334.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import jax
+import jax.config
+jax.config.update("jax_enable_x64", True)
+import jax.numpy as np
+import tensornetwork
+from tensornetwork import contractors
+
+
+@jax.jit
+def binary_mera_energy(hamiltonian, state, isometry, disentangler):
+  """Computes the energy using a layer of uniform binary MERA.
+
+  Args:
+    hamiltonian: The hamiltonian (rank-6 tensor) defined at the bottom of the
+      MERA layer.
+    state: The 3-site reduced state (rank-6 tensor) defined at the top of the
+      MERA layer.
+    isometry: The isometry tensor (rank 3) of the binary MERA.
+    disentangler: The disentangler tensor (rank 4) of the binary MERA.
+
+  Returns:
+    The energy.
+  """
+  out = []
+  for dirn in ('left', 'right'):
+    net = tensornetwork.TensorNetwork(backend="jax")
+
+    iso_l = net.add_node(isometry)
+    iso_c = net.add_node(isometry)
+    iso_r = net.add_node(isometry)
+
+    iso_l_con = net.add_node(np.conj(isometry))
+    iso_c_con = net.add_node(np.conj(isometry))
+    iso_r_con = net.add_node(np.conj(isometry))
+
+    op = net.add_node(hamiltonian)
+    rho = net.add_node(state)
+
+    un_l = net.add_node(disentangler)
+    un_l_con = net.add_node(np.conj(disentangler))
+
+    un_r = net.add_node(disentangler)
+    un_r_con = net.add_node(np.conj(disentangler))
+
+    net.connect(iso_l[2], rho[0])
+    net.connect(iso_c[2], rho[1])
+    net.connect(iso_r[2], rho[2])
+
+    net.connect(iso_l[0], iso_l_con[0])
+    net.connect(iso_l[1], un_l[2])
+    net.connect(iso_c[0], un_l[3])
+    net.connect(iso_c[1], un_r[2])
+    net.connect(iso_r[0], un_r[3])
```

Comment:
`edge1 ^ edge2` is supported now. Feel free to use it. (We're still working on getting pylint to not complain about it though...)